appservice/src/utils/cpUtils.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import * as os from 'os';
+import { CommandOptions } from '../CommandOptions';
+import { localize } from '../localize';
+
+export namespace cpUtils {
+    export async function executeCommand(commandOptions: CommandOptions): Promise<string> {
+        const command: string = commandOptions.command;
+        const result: ICommandResult = await tryExecuteCommand(commandOptions);
+        // command is only being used to output at this point so we can alter it
+        if (result.code !== 0) {
+            // We want to make sure the full error message is displayed to the user, not just the error code.
+            // If outputChannel is defined, then we simply call 'outputChannel.show()' and throw a generic error telling the user to check the output window
+            // If outputChannel is _not_ defined, then we include the command's output in the error itself and rely on AzureActionHandler to display it properly
+
+            if (commandOptions.showOutputChannel()) {
+                throw new Error(localize('commandErrorWithOutput', 'Failed to run "{0}" command. Check output window for more details.', command));
```

Comment:
`command` could have a value that needs to be obfuscated

Reply:
Well, I refactored to have command only have the command and args to have values that should be obfuscated.